plugins/eos/gs-plugin-eos.c

```diff
@@ -2190,27 +2190,42 @@ gs_plugin_add_popular (GsPlugin *plugin,
 {
 	g_autoptr(AsProfileTask) ptask = NULL;
 	GsAppList *new_list = NULL;
+	/* These IDs must match the <component><id> element in the app's
```

Comment:
I wonder if there is a better function to dig in the cached apps to match without hard-coding whether or not the app ID has a .desktop in? https://github.com/endlessm/gnome-software/blob/master/plugins/core/gs-plugin-hardcoded-blacklist.c seems to do some globbing for example.

Reply:
Whether or not there is a .desktop isn't (supposed to be) mutable - the app ID shouldn't change unless there is some weirdness going on. So we don't need to worry about being smarter here, just maintaining this list accurately.